### PR TITLE
chore: Multiple Env code split for 1-click 

### PR DESCRIPTION
**Pull request by ankitsrivas14**
Multiple Env code split for 1-click 
Moved the existing datasourceEnvEnabled selectors to feature flag helpers as isMultipleEnvEnabled

>
#### Type of change

- Chore (housekeeping or task changes that don't impact user perception)

>
>
- [ ] Manual
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

**Review by dipyamanbiswas07 — APPROVED** [review]
LGTM

File: app/client/cypress/support/Pages/MultipleEnvironments.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
+import { agHelper, dataSources } from "../Objects/ObjectsCore";
```
Comment (ayushpahwa) [inline app/client/cypress/support/Pages/MultipleEnvironments.ts:1]: Why are we moving this file to EE?

Comment (ankitsrivas14) [reply app/client/cypress/support/Pages/MultipleEnvironments.ts:1]: This file is already there in EE, here we are moving it to CE as it is in the common file structure. 
All the common files should come from CE to EE, right?
Today, this file is being used in EE only. But I think Multiple Env ramps are not covered in CE cypress today. So whenever we plan to do that this file would be required in CE.

**Review by ankitakinger — APPROVED** [review]

**Review by dipyamanbiswas07 — DISMISSED** [review]
LGTM

**Review by ankitakinger — DISMISSED** [review]

**Review by dipyamanbiswas07 — DISMISSED** [review]
LGTM

**Review by dipyamanbiswas07 — COMMENTED** [review]
LGTM

**Review by dipyamanbiswas07 — APPROVED** [review]

**Pull request merged**